src/rs.cpp

```diff
@@ -88,34 +89,53 @@ struct rs2_device_list
 struct rs2_option_value_wrapper : rs2_option_value
 {
     // Keep the original json value, so we can refer to it (e.g., with as_string)
-    std::shared_ptr< const rsutils::json > p_json;
+    std::shared_ptr< const json > p_json;
 
     // Add a reference count to control lifetime
     mutable std::atomic< int > ref_count;
 
-    rs2_option_value_wrapper( rs2_option option_id, std::shared_ptr< const rsutils::json > const & p_json_value )
+    rs2_option_value_wrapper( rs2_option option_id,
+                              rs2_option_type option_type,
+                              std::shared_ptr< const json > const & p_json_value )
         : ref_count( 1 )
         , p_json( p_json_value )
     {
         id = option_id;
-        type = RS2_OPTION_TYPE_COUNT;
-        if( p_json )
+        type = option_type;
+        if( ! p_json || p_json->is_null() )
         {
-            if( p_json->is_number_float() )
+            is_valid = false;
+        }
+        else
+        {
+            switch( type )
             {
-                type = RS2_OPTION_TYPE_FLOAT;
+            case RS2_OPTION_TYPE_FLOAT:
+                if( ! p_json->is_number() )
+                    throw invalid_value_exception( get_string( option_id )
```

Comment:
Shouldn't we set `is_valid` to `false` before throwing? We can set it in the initialization list.

Reply:
You're right; I though I had set is_valid to false at the beginning for everything, but now I see it's only if no `p_json`... weird.